Common/src/main/java/at/petrak/hexcasting/api/casting/mishaps/MishapNotEnoughMedia.kt

```diff
@@ -0,0 +1,21 @@
+package at.petrak.hexcasting.api.casting.mishaps
+
+import at.petrak.hexcasting.api.casting.eval.CastingEnvironment
+import at.petrak.hexcasting.api.casting.eval.ResolvedPatternType
+import at.petrak.hexcasting.api.casting.iota.Iota
+import at.petrak.hexcasting.api.pigment.FrozenPigment
+import at.petrak.hexcasting.api.utils.asTranslatedComponent
+import net.minecraft.world.item.DyeColor
+
+class MishapNotEnoughMedia(private val cost: Long) : Mishap() {
+    override fun accentColor(ctx: CastingEnvironment, errorCtx: Context): FrozenPigment =
+        dyeColor(DyeColor.RED)
+
+    override fun resolutionType(ctx: CastingEnvironment) = ResolvedPatternType.ERRORED
+
+    override fun execute(env: CastingEnvironment, errorCtx: Context, stack: MutableList<Iota>) {
+        env.extractMedia(cost, false)
+    }
+
+    override fun errorMessage(ctx: CastingEnvironment, errorCtx: Context) = "hexcasting.message.cant_overcast".asTranslatedComponent
```

Comment:
should this lang key be changed to be with the other mishap keys?

Reply:
maybe. I didn't touch lang since it was not strictly required here

Reply:
probably out of scope for now